lib/auth/auth_with_roles.go

```diff
@@ -4587,6 +4587,11 @@ func (a *ServerWithRoles) UpsertTrustedCluster(ctx context.Context, tc types.Tru
 		return nil, trace.Wrap(err)
 	}
 
+	// Don't allow trusted clusters if running in Cloud.
+	if modules.GetModules().Features().Cloud {
+		return nil, trace.NotImplemented("cloud clusters do not support trusted cluster resources")
+	}
+
```

Comment:
This would prevent someone from connecting a cloud cluster to another cluster as a leaf cluster right? What we want to warn/error on is when a leaf cluster is connected to cloud where cloud is the root.